### DISCUSSION TITLE
Change the paging system of docks for something more accessible

### DISCUSSION THREAD

**Discussion by eon-s** [godotengine/godot-proposals#9558] [Editor]
The dock paging system is not good and with the addition of the undo panel (History) everything got too cramped.

Where is my Import panel, is here? was it somewhere else? how would I know without looking everywhere in this paging system?
![image](https://github.com/godotengine/godot-proposals/assets/10215987/82dfb055-66ec-47a8-823f-4edb817f1295)

So, instead of that old and not very useful (other than to use on small screens) paging system, panels could have other ways to be added that takes less space and are still accessible:

- These could be folded (and stacked) instead (similar to inspector sections)
- Folded to the side
- Always visible but with vertical labels like Blender
![image](https://github.com/godotengine/godot-proposals/assets/10215987/f44e9be0-b715-4cf0-8d8f-b10a9aea2ad3)
(with optional icon-only view, again like Blender)
![image](https://github.com/godotengine/godot-proposals/assets/10215987/c0237199-8652-4576-8b65-47d61ca639e7)

Any of these changes to the current paging system will reduce the problems and will make less painful having lot of addons with custom panels.

**Comment by passivestar** [comment]
The tabbar could also be made scrollable with mousewheel. That will make it a little easier to use even with horizontal tabs

**Reply by eon-s** [reply]
It would make it easier to use, yes (and may be easy to add), but will still be a usability pain when you have 2 or 3 addons that use docks.

**Comment by aXu-AP** [comment]
This is something I've run myself too, not in the editor but in an app which was made in Godot. I suggest having an option to shrink the tabs to a certain point, much like browsers do.
